nova/virt/vmwareapi/driver.py

```diff
@@ -753,6 +760,290 @@ def _server_group_sync_loop(self, compute_host):
+                          "for instance: %s", instance["name"],
+                          instance=instance)
+                recover_method(context, instance, dest, migrate_data)
+
+    def _get_volume_mappings(self, context, instance):
+        """Returns a mapping for all the devices with volumes to
+        their connection_info.data, where possible.
+        It is up to the caller to verify that the required information
+        is available.
+        It uses the _current_ block-device mapping.
+        """
+        volume_infos = []
+        for bdm in objects.BlockDeviceMappingList.get_by_instance_uuid(
+                    context, instance.uuid):
+            if not bdm.is_volume:
+                continue
+            # If we have volume_id, `map_volumes_to_devices` will map it to
+            # a device.
+            data = {"volume_id": bdm.volume_id}
+            try:
+                connection_info = jsonutils.loads(bdm.connection_info)
+                data.update(connection_info.get("data", {}))
+            except ValueError as e:
+                message = ("Could not parse connection_info for volume {}."
+                    " Reason: {}"
+                ).format(bdm.volume_id, e)
+                LOG.warning(message, instance=instance)
+
+            # Normalise the datastore reference
+            # As it depends on the caller, if actually need the
+            # value, we do not raise an error here
+            datastore = data.get("datastore") or \
+                            data.get("ds_ref_val")
+            if datastore:
+                data["datastore_ref"] = vim_util.get_moref(datastore,
+                                                        "Datastore")
+            volume_infos.append(data)
+        return self._volumeops.map_volumes_to_devices(instance, volume_infos)
+
+    def _set_vif_infos(self, migrate_data, session):
+        if 'vifs' in migrate_data and migrate_data.vifs:
+            cluster_ref = vim_util.get_moref(migrate_data.dest_cluster_ref,
+                "ClusterComputeResource")
+            dest_network_info = [
+                vif.get_dest_vif()
+                for vif in migrate_data.vifs
+            ]
+
+            vif_model = None  # We are not reading that value
+            migrate_data.vif_infos = vmwarevif.get_vif_info(session,
+                cluster_ref, utils.is_neutron(), vif_model, dest_network_info)
+        else:
+            migrate_data.vif_infos = []
+
+    def _create_dest_session(self, migrate_data):
+        defaults = migrate_data.relocate_defaults
+        service = defaults["service"]
+        credentials = service["credentials"]
+        dest_url = urllib.parse.urlparse(service["url"])
+
+        dest_session = VMwareAPISession(
+            host_ip=dest_url.hostname,
+            host_port=dest_url.port,
+            username=credentials["username"],
+            password=credentials["password"],
+            # TODO(fwiesel): SSL Settings
```

Comment:
You still have TODO here. Does it automatically determine that it has to use SSL?